File: portscan.py
import socket
import sys


if (len(sys.argv) < 2):
    print("How to use portscan")
    print(sys.argv[0], "10.1.1.1")
else:
    for Port in range(1, 65535):
        mysocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        if (mysocket.connect_ex((sys.argv[1], Port)) == 0):
            print("Port ", Port, "[OPENED]")

File: portscan.c
#include <stdio.h>
#include <netdb.h>
int main(int argc, char *argv[])
{
	int meusocket;
	int conecta;
	int port;
	int inicio = 0;
	int final  = 65535;
	char * destino;
	if(argc < 2)
	{
		printf("How to use portscan\n");
		printf("%s 10.1.1.101\n",argv[0]);
		return 0;	
	}

	destino = argv[1];
	struct sockaddr_in alvo;
	
	for(port = inicio; port < final; port++)
	{
		meusocket = socket(AF_INET, SOCK_STREAM, 0);
		alvo.sin_family = AF_INET;
		alvo.sin_port = htons(port);
		alvo.sin_addr.s_addr = inet_addr(destino);
	
		conecta = connect(meusocket, (struct sockaddr *)&alvo, sizeof alvo);
		if(conecta == 0)
		{
			printf("Porta %d - status [ABERTA] \n", port);
		}
		close(meusocket);
		close(conecta);
	}
}
